### PR TITLE
STORY#590_DashboardFlow

### PR DESCRIPTION
**Pull request by rejneesh1**

### CODE REVIEW

File: src/app/app.component.ts [hunk 1]
```diff
@@ -406,4 +410,31 @@ export class MyApp {
   closeModal() {
     this.appUpdateData = null;
   }
+
+  onDashboardClick() {
+    this.localStorage
+      .getLocalStorage("profileRole")
```
Comment (kiranharidas187) [inline src/app/app.component.ts]: put in local storage  constants

File: src/pages/bottom-tab/bottom-tab.ts [hunk 2]
```diff
@@ -21,11 +30,52 @@ export class BottomTabPage {
...
+      (showDashboard) => {
+        this.showReport = showDashboard;
+        this.getProfileroles();
+      }
+    );
+  }
+
+  getProfileroles() {
+    this.localStorage
+      .getLocalStorage("profileRole")
```
Comment (kiranharidas187) [inline src/pages/bottom-tab/bottom-tab.ts]: put in contestants key

File: src/pages/reports/report-entity-listing/report-entity-listing.html [hunk 3]
```diff
@@ -5,13 +5,18 @@
   Ionic pages and navigation.
 -->
 <ion-header>
-  <header [title]="'headings.'+ currentEntityType |translate" [hideBack]="false">
-  </header>
+  <header
+    [title]="'headings.'+ currentEntityType |translate"
```
Comment (kiranharidas187) [inline src/pages/reports/report-entity-listing/report-entity-listing.html]: This you need not translate,, you can use currentEntityType directly

File: src/providers/sidemenu/sidemenu.ts [hunk 4]
```diff
@@ -39,9 +39,12 @@ export class SidemenuProvider {
     let currentUser = this.currentUser.getCurrentUserData();
     this.apiProvider.httpGet(
       AppConfigs.roles.getProfile + currentUser.sub,
-      (success) => {
+      async (success) => {
         this.profileRoles = success.result;
-        this.localStorage.setLocalStorage("profileRole", this.profileRoles);
+        await this.localStorage.setLocalStorage(
+          "profileRole",
```
Comment (kiranharidas187) [inline src/providers/sidemenu/sidemenu.ts:45]: use constant file

File: src/app/app.component.ts (hunk 1 above)
Comment (rejneesh1) [reply src/app/app.component.ts]: done

File: src/pages/bottom-tab/bottom-tab.ts (hunk 2 above)
Comment (rejneesh1) [reply src/pages/bottom-tab/bottom-tab.ts]: done

File: src/pages/reports/report-entity-listing/report-entity-listing.html (hunk 3 above)
Comment (rejneesh1) [reply src/pages/reports/report-entity-listing/report-entity-listing.html]: ok

File: src/providers/sidemenu/sidemenu.ts (hunk 4 above)
Comment (rejneesh1) [reply src/providers/sidemenu/sidemenu.ts:45]: changed

**Pull request merged**